src/components/Button/styles.tsx

```diff
@@ -0,0 +1,19 @@
+import styled from "styled-components";
+import { theme } from "~/styles/defaultTheme";
+
+export const ButtonComponent = styled.button`
+  width: 15em;
+  height: 3em;
+  font-size: large;
+  margin: 1em;
+  padding: 10px 10px;
+  color: white;
+  background-color: ${theme.colors.$primary};
```

Comment:
use theme from inside styled component:

`background-color: ${({theme}) => theme.colors.$primary}`

Also using $ before colors is not a good practice in styled components, you can change for theme.colors.primary